// @tencentcloud/roomkit-web-vue3@2.9.1/es/components/Stream/SingleStreamView/index.vue.mjs
import { defineComponent, ref, createBlock, createCommentVNode, openBlock, unref } from "vue";
import StreamRegion from "../common/StreamRegion/StreamRegionPC.vue.mjs";
import { isPC } from "../../../utils/environment.mjs";
import { StreamPlayQuality } from "../../../services/manager/mediaManager.mjs";
const _sfc_main = /* @__PURE__ */ defineComponent({
  __name: "index",
  props: {
    streamInfo: {},
    fillMode: {},
    supportTouchScale: { type: Boolean }
  },
  emits: ["stream-view-dblclick"],
  setup(__props) {
    const aspectRatio = ref(isPC ? "16:9" : "auto");
    return (_ctx, _cache) => {
      return _ctx.streamInfo ? (openBlock(), createBlock(unref(StreamRegion), {
        key: 0,
        "stream-info": _ctx.streamInfo,
        "fill-mode": _ctx.fillMode,
        "support-touch-scale": _ctx.supportTouchScale,
        "stream-play-quality": unref(StreamPlayQuality).HIGH,
        "aspect-ratio": aspectRatio.value,
        onStreamViewDblclick: _cache[0] || (_cache[0] = ($event) => _ctx.$emit("stream-view-dblclick"))
      }, null, 8, ["stream-info", "fill-mode", "support-touch-scale", "stream-play-quality", "aspect-ratio"])) : createCommentVNode("", true);
    };
  }
});
export {
  _sfc_main as default
};
