// @tencentcloud/roomkit-web-vue3@2.9.1/es/components/common/base/Timepicker/TimepickerPC.vue.mjs
import { defineComponent, ref, computed, watch, createElementBlock, openBlock, createVNode, unref, withCtx, Fragment, renderList, createBlock } from "vue";
import Select from "../Select/index.mjs";
import OPtion from "../Option/index.mjs";
const _hoisted_1 = { class: "container" };
const _sfc_main = /* @__PURE__ */ defineComponent({
  __name: "TimepickerPC",
  props: {
    modelValue: {}
  },
  emits: ["input"],
  setup(__props, { emit: __emit }) {
    const props = __props;
    const emit = __emit;
    const selectedTime = ref(props.modelValue);
    const timeOptions = computed(() => {
      const options = [];
      for (let i = 0; i < 24; i++) {
        for (let j = 0; j < 60; j += 15) {
          const hour = i < 10 ? `0${i}` : `${i}`;
          const minute = j < 10 ? `0${j}` : `${j}`;
          options.push(`${hour}:${minute}`);
        }
      }
      return options;
    });
    const updateTime = () => {
      emit("input", selectedTime.value);
    };
    watch(
      selectedTime,
      () => {
        updateTime();
      },
      {
        immediate: true
      }
    );
    watch(
      () => props.modelValue,
      (newValue) => {
        selectedTime.value = newValue;
      },
      { immediate: true }
    );
    return (_ctx, _cache) => {
      return openBlock(), createElementBlock("div", _hoisted_1, [
        createVNode(unref(Select), {
          modelValue: selectedTime.value,
          "onUpdate:modelValue": _cache[0] || (_cache[0] = ($event) => selectedTime.value = $event),
          class: "timepicker-select",
          teleported: false,
          "custom-select-content-style": { "font-weight": 400 }
        }, {
          default: withCtx(() => [
            (openBlock(true), createElementBlock(Fragment, null, renderList(timeOptions.value, (time) => {
              return openBlock(), createBlock(unref(OPtion), {
                key: time,
                value: time,
                label: time,
                "custom-option-content-style": { "font-weight": 400 }
              }, null, 8, ["value", "label"]);
            }), 128))
          ]),
          _: 1
        }, 8, ["modelValue"])
      ]);
    };
  }
});
export {
  _sfc_main as default
};
