// @tencentcloud/roomkit-web-vue3@2.9.1/lib/hooks/useCustomizedAutoPlayDialog.js
"use strict";
Object.defineProperties(exports, { __esModule: { value: true }, [Symbol.toStringTag]: { value: "Module" } });
const TUIRoomEngine = require("@tencentcloud/tuiroom-engine-js");
const useRoomEngine = require("./useRoomEngine.js");
const index$1 = require("../components/common/base/MessageBox/index.js");
const index = require("../locales/index.js");
const roomEngine = useRoomEngine.default();
let isShowAutoPlayDialog = false;
function useCustomizedAutoPlayDialog() {
  const { t } = index.useI18n();
  TUIRoomEngine.once("ready", () => {
    var _a;
    const trtcCloud = (_a = roomEngine.instance) == null ? void 0 : _a.getTRTCCloud();
    trtcCloud.callExperimentalAPI(
      JSON.stringify({
        api: "enableAutoPlayDialog",
        params: { enable: 0 }
      })
    );
    trtcCloud.on("onAutoPlayFailed", () => {
      if (!isShowAutoPlayDialog) {
        isShowAutoPlayDialog = true;
        index$1.default({
          title: t("Attention"),
          message: t(
            'Audio playback failed. Click the "Confirm" to resume playback'
          ),
          confirmButtonText: t("Confirm"),
          callback: () => {
            isShowAutoPlayDialog = false;
          }
        });
      }
    });
  });
}
exports.default = useCustomizedAutoPlayDialog;
