// @tencentcloud/roomkit-web-vue3@2.9.1/lib/components/common/ArrowStroke.vue2.js
"use strict";
Object.defineProperties(exports, { __esModule: { value: true }, [Symbol.toStringTag]: { value: "Module" } });
const Vue = require("vue");
const SvgIcon = require("./base/SvgIcon.vue.js");
const ArrowStrokeLeftIcon = require("./icons/ArrowStrokeLeftIcon.vue.js");
const _sfc_main = /* @__PURE__ */ Vue.defineComponent({
  __name: "ArrowStroke",
  props: {
    strokePosition: {},
    arrowDirection: {},
    hasStroke: { type: Boolean }
  },
  emits: ["click-arrow"],
  setup(__props, { emit: __emit }) {
    const props = __props;
    const emits = __emit;
    const arrowStrokeClass = Vue.computed(
      () => `arrow-stroke-container-${props.strokePosition}`
    );
    function handleClickArrow() {
      emits("click-arrow");
    }
    return (_ctx, _cache) => {
      return Vue.openBlock(), Vue.createElementBlock("div", {
        class: Vue.normalizeClass(arrowStrokeClass.value)
      }, [
        Vue.createElementVNode("div", {
          class: Vue.normalizeClass(["stroke-content", { "has-stroke": _ctx.hasStroke }])
        }, [
          Vue.createElementVNode("div", {
            class: "arrow-content",
            onClick: handleClickArrow
          })
        ], 2),
        Vue.createVNode(SvgIcon.default, {
          class: Vue.normalizeClass(["arrow", { "has-stroke": _ctx.hasStroke }]),
          onClick: handleClickArrow
        }, {
          default: Vue.withCtx(() => [
            Vue.createVNode(ArrowStrokeLeftIcon.default, {
              class: Vue.normalizeClass(_ctx.arrowDirection)
            }, null, 8, ["class"])
          ]),
          _: 1
        }, 8, ["class"])
      ], 2);
    };
  }
});
exports.default = _sfc_main;
