// @tencentcloud/roomkit-web-vue3@2.9.1/es/components/common/widgets/Notification/InviteNotification/index.vue2.mjs
import { defineComponent, ref, watch, onMounted, withDirectives, createElementBlock, openBlock, normalizeClass, createElementVNode, normalizeStyle, createVNode, toDisplayString, unref, withCtx, vShow } from "vue";
import TuiButton from "../../../base/Button.vue.mjs";
import SvgIcon from "../../../base/SvgIcon.vue.mjs";
import useZIndex from "../../../../../hooks/useZIndex.mjs";
import EnterRoomIcon from "../../../icons/EnterRoomIcon.vue.mjs";
import { useI18n } from "../../../../../locales/index.mjs";
import TuiAvatar from "../../../Avatar.vue.mjs";
const _hoisted_1 = { class: "tui-notification-message" };
const _hoisted_2 = { class: "invitation-notification-inviter-container" };
const _hoisted_3 = { class: "invitation-notification-inviter" };
const _hoisted_4 = { class: "invitation-notification-title" };
const _hoisted_5 = { class: "invitation-notification-room-name" };
const _hoisted_6 = { class: "invitation-notification-room-info-container" };
const _hoisted_7 = { class: "invitation-notification-room-text-container" };
const _hoisted_8 = { class: "invitation-notification-room-text" };
const _hoisted_9 = { class: "invitation-notification-room-text" };
const _hoisted_10 = { class: "invitation-notification-room-text" };
const _hoisted_11 = { class: "invitation-notification-room-text" };
const _hoisted_12 = { class: "invitation-notification-bottom-container" };
const _hoisted_13 = { class: "button-text" };
const _sfc_main = /* @__PURE__ */ defineComponent({
  __name: "index",
  props: {
    appendTo: { default: "" },
    customClass: { default: "" },
    message: { default: {
      userId: "",
      userName: "",
      avatarUrl: "",
      roomName: "",
      roomMemberCount: 0,
      roomOwner: "",
      roomId: ""
    } },
    onConfirm: { type: [Function, null], default: null },
    onCancel: { type: [Function, null], default: null },
    close: { type: [Function, null], default: null },
    confirmButtonText: { default: "" },
    cancelButtonText: { default: "" }
  },
  setup(__props) {
    const { t } = useI18n();
    const visible = ref(false);
    const overlayContainerStyle = ref({});
    const { nextZIndex } = useZIndex();
    const props = __props;
    watch(visible, (val) => {
      if (val) {
        overlayContainerStyle.value = { zIndex: nextZIndex() };
      }
    });
    function onOpen() {
      visible.value = true;
    }
    const handleConfirm = () => {
      handleAction(props.onConfirm);
    };
    const handleCancel = () => {
      handleAction(props.onCancel);
    };
    const handleAction = (fn) => {
      if (fn) {
        fn();
      }
      if (props.close) {
        props.close();
      }
    };
    onMounted(() => {
      onOpen();
    });
    return (_ctx, _cache) => {
      return withDirectives((openBlock(), createElementBlock("div", {
        class: normalizeClass(["tui-notification", _ctx.customClass])
      }, [
        createElementVNode("div", {
          class: "tui-notification-container",
          style: normalizeStyle(overlayContainerStyle.value)
        }, [
          createElementVNode("div", _hoisted_1, [
            createElementVNode("div", _hoisted_2, [
              createElementVNode("div", _hoisted_3, [
                createVNode(TuiAvatar, {
                  class: "avatar",
                  "img-src": _ctx.message.avatarUrl
                }, null, 8, ["img-src"]),
                createElementVNode("span", _hoisted_4, toDisplayString(unref(t)("sb invites you to join the conference", {
                  name: _ctx.message.userName || _ctx.message.userId
                })), 1)
              ]),
              createElementVNode("span", _hoisted_5, toDisplayString(_ctx.message.roomName), 1),
              createElementVNode("div", _hoisted_6, [
                createElementVNode("div", _hoisted_7, [
                  createElementVNode("span", _hoisted_8, toDisplayString(unref(t)("Host")) + ": ", 1),
                  createElementVNode("span", _hoisted_9, toDisplayString(_ctx.message.roomOwner), 1)
                ]),
                _cache[0] || (_cache[0] = createElementVNode("i", { class: "invitation-notification-room-text-split" }, null, -1)),
                createElementVNode("div", null, [
                  createElementVNode("span", _hoisted_10, toDisplayString(unref(t)("Attendees")) + ": ", 1),
                  createElementVNode("span", _hoisted_11, toDisplayString(`${_ctx.message.roomMemberCount} ${unref(t)("people")}`), 1)
                ])
              ])
            ])
          ]),
          _cache[1] || (_cache[1] = createElementVNode("i", { class: "invitation-notification-bottom-split" }, null, -1)),
          createElementVNode("div", _hoisted_12, [
            createElementVNode("span", {
              class: "invitation-notification-bottom",
              onClick: handleCancel
            }, toDisplayString(_ctx.cancelButtonText), 1),
            createVNode(TuiButton, {
              class: "button-item",
              style: { "width": "110px" },
              onClick: handleConfirm
            }, {
              default: withCtx(() => [
                createVNode(SvgIcon, {
                  class: "icon",
                  icon: EnterRoomIcon
                }),
                createElementVNode("span", _hoisted_13, toDisplayString(_ctx.confirmButtonText), 1)
              ]),
              _: 1
            })
          ])
        ], 4)
      ], 2)), [
        [vShow, visible.value]
      ]);
    };
  }
});
export {
  _sfc_main as default
};
