// @tencentcloud/roomkit-web-vue3@2.9.1/lib/components/common/base/Message/images/info.vue.js
"use strict";
Object.defineProperties(exports, { __esModule: { value: true }, [Symbol.toStringTag]: { value: "Module" } });
const Vue = require("vue");
const _pluginVue_exportHelper = require("../../../../../_virtual/_plugin-vue_export-helper.js");
const _sfc_main = {};
const _hoisted_1 = {
  width: "20",
  height: "20",
  viewBox: "0 0 20 20",
  fill: "none",
  xmlns: "http://www.w3.org/2000/svg"
};
function _sfc_render(_ctx, _cache) {
  return Vue.openBlock(), Vue.createElementBlock("svg", _hoisted_1, _cache[0] || (_cache[0] = [
    Vue.createElementVNode("g", { id: "提示" }, [
      Vue.createElementVNode("path", {
        id: "Vector",
        "fill-rule": "evenodd",
        "clip-rule": "evenodd",
        d: "M10.0007 17.4992C14.1428 17.4992 17.5006 14.1414 17.5006 9.99919C17.5006 5.85705 14.1428 2.49919 10.0007 2.49919C5.85852 2.49919 2.50065 5.85705 2.50065 9.99919C2.50065 14.1414 5.85852 17.4992 10.0007 17.4992ZM19.1673 9.99919C19.1673 15.0618 15.0632 19.1659 10.0007 19.1659C4.93804 19.1659 0.833984 15.0618 0.833984 9.99919C0.833984 4.93658 4.93804 0.83252 10.0007 0.83252C15.0632 0.83252 19.1673 4.93658 19.1673 9.99919ZM9.16732 14.5825V8.33252H10.834V14.5825H9.16732ZM10.834 7.08252H9.16407V5.4126H10.834V7.08252Z",
        fill: "currentColor"
      })
    ], -1)
  ]));
}
const info = /* @__PURE__ */ _pluginVue_exportHelper.default(_sfc_main, [["render", _sfc_render]]);
exports.default = info;
