// @tencentcloud/roomkit-web-vue3@2.9.1/lib/constants/room.js
"use strict";
Object.defineProperty(exports, Symbol.toStringTag, { value: "Module" });
const TUIRoomEngine = require("@tencentcloud/tuiroom-engine-js");
var IconButtonLayout = /* @__PURE__ */ ((IconButtonLayout2) => {
  IconButtonLayout2["VERTICAl"] = "vertical";
  IconButtonLayout2["HORIZONTAL"] = "horizontal";
  return IconButtonLayout2;
})(IconButtonLayout || {});
({
  videoResolution: TUIRoomEngine.TRTCVideoResolution.TRTCVideoResolution_640_360
});
var USERS_STATUS = /* @__PURE__ */ ((USERS_STATUS2) => {
  USERS_STATUS2[USERS_STATUS2["Entered"] = 1] = "Entered";
  USERS_STATUS2[USERS_STATUS2["ON_STAGE"] = 1] = "ON_STAGE";
  USERS_STATUS2[USERS_STATUS2["NOT_ON_STAGE"] = 2] = "NOT_ON_STAGE";
  USERS_STATUS2[USERS_STATUS2["NOT_ENTER"] = 3] = "NOT_ENTER";
  return USERS_STATUS2;
})(USERS_STATUS || {});
const PASSWORD_MAX_LENGTH_LIMIT = 6;
var TRTCBeautyStyle = /* @__PURE__ */ ((TRTCBeautyStyle2) => {
  TRTCBeautyStyle2[TRTCBeautyStyle2["TRTCBeautyStyleSmooth"] = 0] = "TRTCBeautyStyleSmooth";
  TRTCBeautyStyle2[TRTCBeautyStyle2["TRTCBeautyStyleNature"] = 1] = "TRTCBeautyStyleNature";
  return TRTCBeautyStyle2;
})(TRTCBeautyStyle || {});
exports.IconButtonLayout = IconButtonLayout;
exports.PASSWORD_MAX_LENGTH_LIMIT = PASSWORD_MAX_LENGTH_LIMIT;
exports.TRTCBeautyStyle = TRTCBeautyStyle;
exports.USERS_STATUS = USERS_STATUS;
