// @tencentcloud/roomkit-web-vue3@2.9.1/es/components/common/ArrowStroke.vue2.mjs
import { defineComponent, computed, createElementBlock, openBlock, normalizeClass, createElementVNode, createVNode, withCtx } from "vue";
import SvgIcon from "./base/SvgIcon.vue.mjs";
import ArrowStrokeLeftIcon from "./icons/ArrowStrokeLeftIcon.vue.mjs";
const _sfc_main = /* @__PURE__ */ defineComponent({
  __name: "ArrowStroke",
  props: {
    strokePosition: {},
    arrowDirection: {},
    hasStroke: { type: Boolean }
  },
  emits: ["click-arrow"],
  setup(__props, { emit: __emit }) {
    const props = __props;
    const emits = __emit;
    const arrowStrokeClass = computed(
      () => `arrow-stroke-container-${props.strokePosition}`
    );
    function handleClickArrow() {
      emits("click-arrow");
    }
    return (_ctx, _cache) => {
      return openBlock(), createElementBlock("div", {
        class: normalizeClass(arrowStrokeClass.value)
      }, [
        createElementVNode("div", {
          class: normalizeClass(["stroke-content", { "has-stroke": _ctx.hasStroke }])
        }, [
          createElementVNode("div", {
            class: "arrow-content",
            onClick: handleClickArrow
          })
        ], 2),
        createVNode(SvgIcon, {
          class: normalizeClass(["arrow", { "has-stroke": _ctx.hasStroke }]),
          onClick: handleClickArrow
        }, {
          default: withCtx(() => [
            createVNode(ArrowStrokeLeftIcon, {
              class: normalizeClass(_ctx.arrowDirection)
            }, null, 8, ["class"])
          ]),
          _: 1
        }, 8, ["class"])
      ], 2);
    };
  }
});
export {
  _sfc_main as default
};
