// @tencentcloud/roomkit-web-vue3@2.9.1/lib/utils/common/logger/index.js
"use strict";
Object.defineProperties(exports, { __esModule: { value: true }, [Symbol.toStringTag]: { value: "Module" } });
const loggerConstants = require("./logger-constants.js");
const loggerUtils = require("./logger-utils.js");
let currentLogLevel = loggerUtils.LogLevelType.LOG_LEVEL_DEBUG;
const logger = {
  debug(...args) {
    if (currentLogLevel <= loggerUtils.LogLevelType.LOG_LEVEL_DEBUG) {
      console.debug(`${loggerUtils.getLogPrefix(loggerUtils.LogContext.RENDER)}`, ...args);
    }
  },
  log(...args) {
    if (currentLogLevel <= loggerUtils.LogLevelType.LOG_LEVEL_LOG) {
      console.log(`${loggerUtils.getLogPrefix(loggerUtils.LogContext.RENDER)}`, ...args);
    }
  },
  info(...args) {
    if (currentLogLevel <= loggerUtils.LogLevelType.LOG_LEVEL_INFO) {
      console.info(`${loggerUtils.getLogPrefix(loggerUtils.LogContext.RENDER)}`, ...args);
    }
  },
  warn(...args) {
    if (currentLogLevel <= loggerUtils.LogLevelType.LOG_LEVEL_WARN) {
      console.warn(`${loggerUtils.getLogPrefix(loggerUtils.LogContext.RENDER)}`, ...args);
    }
  },
  error(...args) {
    if (currentLogLevel <= loggerUtils.LogLevelType.LOG_LEVEL_ERROR) {
      console.error(`${loggerUtils.getLogPrefix(loggerUtils.LogContext.RENDER)}`, ...args);
    }
  },
  setLevel(newLevel) {
    var _a;
    if (newLevel >= loggerUtils.LogLevelType.LOG_LEVEL_DEBUG && newLevel <= loggerUtils.LogLevelType.LOG_LEVEL_NON_LOGGING) {
      console.log(
        `${loggerUtils.getLogPrefix(
          loggerUtils.LogContext.RENDER
        )} set log level from ${currentLogLevel} to ${newLevel}`
      );
      currentLogLevel = newLevel;
      if ((_a = window.electron) == null ? void 0 : _a.ipcRenderer) {
        window.electron.ipcRenderer.send(
          loggerConstants.EUserEventNames.ON_CHANGE_LOG_LEVEL,
          newLevel
        );
      }
    } else {
      console.error(
        `${loggerUtils.getLogPrefix(
          loggerUtils.LogContext.RENDER
        )} logger.setLevel() invalid params:${newLevel}`
      );
    }
  },
  getLevel() {
    return currentLogLevel;
  }
};
exports.default = logger;
