// @tencentcloud/roomkit-web-vue3@2.9.1/lib/components/common/base/PopUpH5.vue2.js
"use strict";
Object.defineProperties(exports, { __esModule: { value: true }, [Symbol.toStringTag]: { value: "Module" } });
const Vue = require("vue");
const SvgIcon = require("./SvgIcon.vue.js");
const basic = require("../../../stores/basic.js");
const ArrowStrokeBackIcon = require("../icons/ArrowStrokeBackIcon.vue.js");
const vTap = require("../../../directives/vTap.js");
const _hoisted_1 = { class: "popup-container" };
const _hoisted_2 = { class: "popup-main-header" };
const _hoisted_3 = { class: "icon-container" };
const _hoisted_4 = { class: "sidebar-title" };
const _hoisted_5 = { class: "popup-main-content" };
const _hoisted_6 = { class: "popup-main-footer" };
const _sfc_main = /* @__PURE__ */ Vue.defineComponent({
  __name: "PopUpH5",
  props: {
    title: {}
  },
  setup(__props) {
    const basicStore = basic.useBasicStore();
    function handleClose() {
      basicStore.setSidebarOpenStatus(false);
      basicStore.setSidebarName("");
    }
    return (_ctx, _cache) => {
      return Vue.openBlock(), Vue.createElementBlock("div", _hoisted_1, [
        Vue.createElementVNode("div", _hoisted_2, [
          Vue.withDirectives((Vue.openBlock(), Vue.createElementBlock("span", _hoisted_3, [
            Vue.createVNode(SvgIcon.default, {
              class: "close-icon",
              icon: ArrowStrokeBackIcon.default
            })
          ])), [
            [Vue.unref(vTap.default), handleClose]
          ]),
          Vue.createElementVNode("span", _hoisted_4, Vue.toDisplayString(_ctx.title), 1)
        ]),
        Vue.createElementVNode("div", _hoisted_5, [
          Vue.renderSlot(_ctx.$slots, "sidebarContent", {}, void 0, true)
        ]),
        Vue.createElementVNode("div", _hoisted_6, [
          Vue.renderSlot(_ctx.$slots, "sidebarFooter", {}, void 0, true)
        ])
      ]);
    };
  }
});
exports.default = _sfc_main;
