// @tencentcloud/roomkit-web-vue3@2.9.1/es/components/common/base/ActionSheep.vue2.mjs
import { defineComponent, ref, watch, createElementBlock, openBlock, withDirectives, createElementVNode, vShow, unref, normalizeStyle, createCommentVNode, renderSlot, createVNode, toDisplayString } from "vue";
import SvgIcon from "./SvgIcon.vue.mjs";
import ArrowDown from "../icons/ArrowDown.vue.mjs";
import vTap from "../../../directives/vTap.mjs";
const _hoisted_1 = { class: "mask" };
const _hoisted_2 = {
  key: 0,
  class: "container-close"
};
const _hoisted_3 = {
  key: 1,
  class: "container-header"
};
const _sfc_main = /* @__PURE__ */ defineComponent({
  __name: "ActionSheep",
  props: {
    visible: {
      type: Boolean,
      default: false
    },
    title: {
      type: String,
      default: ""
    },
    showClose: {
      type: Boolean,
      default: true
    },
    showMask: {
      type: Boolean,
      default: true
    },
    height: {
      type: String,
      default: ""
    }
  },
  emits: ["input", "close"],
  setup(__props, { emit: __emit }) {
    const props = __props;
    const emit = __emit;
    const show = ref(false);
    watch(
      () => props.visible,
      (val) => show.value = val,
      { immediate: true }
    );
    watch(
      show,
      (val) => {
        emit("input", val);
      },
      { immediate: true }
    );
    const toggleShow = () => {
      show.value = !show.value;
      if (!show.value) {
        emit("close");
      }
    };
    return (_ctx, _cache) => {
      return openBlock(), createElementBlock("div", null, [
        withDirectives(createElementVNode("div", _hoisted_1, null, 512), [
          [vShow, show.value && __props.showMask],
          [unref(vTap), toggleShow]
        ]),
        withDirectives(createElementVNode("div", {
          class: "container",
          style: normalizeStyle({ height: props.height })
        }, [
          __props.showClose ? withDirectives((openBlock(), createElementBlock("div", _hoisted_2, [
            createVNode(SvgIcon, { icon: ArrowDown })
          ])), [
            [unref(vTap), toggleShow]
          ]) : createCommentVNode("", true),
          __props.title ? (openBlock(), createElementBlock("div", _hoisted_3, toDisplayString(__props.title), 1)) : createCommentVNode("", true),
          renderSlot(_ctx.$slots, "default", {}, void 0, true)
        ], 4), [
          [vShow, show.value]
        ])
      ]);
    };
  }
});
export {
  _sfc_main as default
};
