// @tencentcloud/roomkit-web-vue3@2.9.1/es/components/common/base/PopUpH5.vue2.mjs
import { defineComponent, createElementBlock, openBlock, createElementVNode, withDirectives, createVNode, unref, toDisplayString, renderSlot } from "vue";
import SvgIcon from "./SvgIcon.vue.mjs";
import { useBasicStore } from "../../../stores/basic.mjs";
import ArrowStrokeBackIcon from "../icons/ArrowStrokeBackIcon.vue.mjs";
import vTap from "../../../directives/vTap.mjs";
const _hoisted_1 = { class: "popup-container" };
const _hoisted_2 = { class: "popup-main-header" };
const _hoisted_3 = { class: "icon-container" };
const _hoisted_4 = { class: "sidebar-title" };
const _hoisted_5 = { class: "popup-main-content" };
const _hoisted_6 = { class: "popup-main-footer" };
const _sfc_main = /* @__PURE__ */ defineComponent({
  __name: "PopUpH5",
  props: {
    title: {}
  },
  setup(__props) {
    const basicStore = useBasicStore();
    function handleClose() {
      basicStore.setSidebarOpenStatus(false);
      basicStore.setSidebarName("");
    }
    return (_ctx, _cache) => {
      return openBlock(), createElementBlock("div", _hoisted_1, [
        createElementVNode("div", _hoisted_2, [
          withDirectives((openBlock(), createElementBlock("span", _hoisted_3, [
            createVNode(SvgIcon, {
              class: "close-icon",
              icon: ArrowStrokeBackIcon
            })
          ])), [
            [unref(vTap), handleClose]
          ]),
          createElementVNode("span", _hoisted_4, toDisplayString(_ctx.title), 1)
        ]),
        createElementVNode("div", _hoisted_5, [
          renderSlot(_ctx.$slots, "sidebarContent", {}, void 0, true)
        ]),
        createElementVNode("div", _hoisted_6, [
          renderSlot(_ctx.$slots, "sidebarFooter", {}, void 0, true)
        ])
      ]);
    };
  }
});
export {
  _sfc_main as default
};
